packaging/el7/charliecloud.spec

```diff
@@ -0,0 +1,91 @@
+Name:           charliecloud
+Version:        VERSION
+Release:        RELEASE%{?dist}
+Summary:        Lightweight user-defined software stacks for high-performance computing
+License:        ASL 2.0
+URL:            https://github.com/hpc/%{name}
+# TODO: update source0 with URL (with pre-built documentation)
+Source0:        %{name}-%{version}.tar.gz
+BuildRequires:  gcc    >= 4.8.5
+BuildRequires:  make   >= 3.82
+
+%package doc
+Summary:        Charliecloud examples and test suite
+Requires:       %{name} = %{version}
+Requires:       bats   >= 0.4.0
+Requires:       bash   >= 4.2.46
+Requires:       wget   >= 1.14
+
+%description
+Charliecloud uses Linux user namespaces to run containers with no privileged
+operations or daemons and minimal configuration changes on center resources.
+This simple approach avoids most security risks while maintaining access to
+the performance and functionality already on offer.
+
+Container images can be built using Docker or anything else that can generate a 
+standard Linux filesystem tree.
+
+For more information visit: https://hpc.github.io/charliecloud/
+
+%description doc
+This package contains the Charliecloud test suite and examples.
+
+# Black magics to stop our python scripts from being byte compiled on Centos7, 
+# which otherwise results in rpmbuild failing to make the package. see:
+# https://github.com/scylladb/scylla/issues/2235
+%global __os_install_post    \
+     /usr/lib/rpm/redhat/brp-compress \
+     %{!?__debug_package:\
+     /usr/lib/rpm/redhat/brp-strip %{__strip} \
+     /usr/lib/rpm/redhat/brp-strip-comment-note %{__strip} %{__objdump} \
+     } \
+     /usr/lib/rpm/redhat/brp-strip-static-archive %{__strip} \
+     %{!?__jar_repack:/usr/lib/rpm/redhat/brp-java-repack-jars} \
+%{nil}
+
+%prep
+%setup -q
+
+%build
+%{__make} %{?mflags}
+
+%install
+%{__make} %{?mflags_install} install PREFIX=%{_prefix} DESTDIR=%{buildroot}
+
+%postun
```

Comment:
This will blow up the very first time you do an upgrade on a charliecloud RPM.  The `%postun` scriptlets don't only run at uninstall time; they also run at **upgrade** time.  You need to check the value of `$1` to see which is which.  See https://fedoraproject.org/wiki/Packaging:Scriptlets#Syntax for a breakdown.

Reply:
> upgrade on a charliecloud RPM

Any way we can test an upgrade with just one RPM (i.e., no lower-versioned package already on the system)?

Reply:
No; you have to install a lower-versioned RPM in order to have it upgrade to a higher-versioned one.  The `%postun` scriptlet can't get run if there's no older RPM being removed that has a `%postun` scriptlet. :-)